src/cms/views/push_notifications/push_notification_sender.py

```diff
@@ -1,43 +1,116 @@
+            push_notification=self.push_notification
+        ).exclude(self.primary_pnt)
+        for secondary_pnt in secondary_pnts:
+            if (
+                secondary_pnt.title == ""
+                and pnt_const.USE_MAIN_LANGUAGE == self.push_notification.mode
+            ):
+                secondary_pnt.title = self.primary_pnt.title
+                secondary_pnt.text = self.primary_pnt.text
+                self.prepared_pnts.append(secondary_pnt)
+            if len(secondary_pnt.title) > 0:
+                self.prepared_pnts.append(secondary_pnt)
+
+    def is_valid(self):
+        """
+        Check if all data for sending push notifications is available
+
+        :return: all prepared push notification translations are valid
+        :rtype: bool
+        """
+        for pnt in self.prepared_pnts:
+            if not pnt.title:
+                self.logger.info("Push Notification Translation invalid: %s", str(pnt))
+                return False
+        return True
+
+    def get_auth_key(self):
+        """
+        Get FCM API auth key
+
+        :return: FCM API auth key
+        :rtype: str
+        """
         fcm_auth_config_key = "fcm_auth_key"
-        auth_key = Configuration.objects.filter(key=fcm_auth_config_key)
-        if auth_key.exists():
+        auth_key = Configuration.objects.filter(key=fcm_auth_config_key).first().value
+        if auth_key:
             self.logger.info("Got fcm_auth_key from database")
-        else:
-            self.logger.info(
-                "Could not get %s from configuration database.", fcm_auth_config_key
-            )
-            return False
-        payload = {
-            "to": f"/topics/{region_slug}-{lan_code}-{channel}",
-            "notification": {"title": title, "body": message},
-            "data": {"lanCode": lan_code, "city": region_slug},
-        }
-        headers = {"Authorization": f"key={auth_key.first().value}"}
-        res = requests.post(self.fcm_url, json=payload, headers=headers)
-        if res.status_code == 200:
-            self.logger.info("Message sent, id: %s", res.json()["message_id"])
-            return True
+            return auth_key
         self.logger.info(
-            "Received invalid response from FCM for push notification: %s, response body: %s",
-            res.status_code,
-            res.text,
+            "Could not get %s from configuration database", fcm_auth_config_key
         )
-        return False
+        return None
+
+    # pylint: disable=too-many-arguments
+    def send(self):
+        """
+        Send all prepared push notification translations
+
+        :return: Success status
+        :rtype: bool
+        """
+        auth_key = self.get_auth_key()
+        status = True
+        for pnt in self.prepared_pnts:
+            payload = {
+                "to": f"/topics/{self.push_notification.region.id}-{pnt.language.code}-{self.push_notification.channel}",
```

Comment:
We defined the channels with the Wordpress Blog ID. This now poses a problem. We have 2 choices: we either import the Wordpress sites in a way that preserves their blog ID in Django. Alternatively, we can add a `blog_id` attribute to our region model.

Reply:
I think it should be possible to import the WordPress blogs in a way that their blog id matches the region id.
Both database fields are identical (unique int, auto increement etc) and we don't have to merge it with other data sets...